inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/LoadNodeUtils.java

```diff
@@ -162,29 +166,58 @@ public static KafkaLoadNode createLoadNode(KafkaSink kafkaSink, List<FieldInfo>
                 format = new JsonFormat();
                 break;
             case CANAL:
-                format = new CanalJsonFormat();
+                format = new RawFormat();
+                innerFormat = new CanalJsonFormat();
                 break;
             case DEBEZIUM_JSON:
-                format = new DebeziumJsonFormat();
+                format = new RawFormat();
+                innerFormat = new DebeziumJsonFormat();
                 break;
             default:
                 throw new IllegalArgumentException(String.format("Unsupported dataType=%s for Kafka", dataType));
         }
 
+        String sinkPartitioner = null;
+        if (dataType == DataTypeEnum.CANAL || dataType == DataTypeEnum.DEBEZIUM_JSON) {
+            sinkPartitioner = kafkaSink.getPartitionStrategy() == null ? null : RAW_HASH;
+        }
+
+        if (StringUtils.isNotEmpty(kafkaSink.getTopicName())) {
+            return new KafkaLoadNode(
+                    kafkaSink.getSinkName(),
+                    kafkaSink.getSinkName(),
+                    fieldInfos,
+                    fieldRelations,
+                    Lists.newArrayList(),
+                    null,
+                    kafkaSink.getTopicName(),
+                    kafkaSink.getBootstrapServers(),
+                    format,
+                    sinkParallelism,
+                    properties,
+                    kafkaSink.getPrimaryKey()
+            );
+        }
+
         return new KafkaLoadNode(
                 kafkaSink.getSinkName(),
                 kafkaSink.getSinkName(),
                 fieldInfos,
                 fieldRelations,
                 Lists.newArrayList(),
                 null,
-                kafkaSink.getTopicName(),
+                "mock_topic",
```

Comment:
Maybe it is not necessary to add a default for the topic?

Reply:
topicname is a nonnull parameter, so this must not be null

Reply:
You can do it in upstream services for now.